Enhance various statements and diagrams

knz opened the pull request:
Fixes #2719.
Fixes #2519.
Fixes #2182.
Fixes #2221.
Fixes #2350.
Fixes #2797.
First step towards fixing #534.

**tl;dr**: fix confusion, better document joins, fix upsert/delete/insert/update.

Prior to this patch, the documentation was confusing "SELECT
statements" and "selection clauses". This confusion was most visible in the documentation for INSERT and UPSERT, which incorrectly specified that these statements take a selection clause as operand, whereas they support much more than that.

The documentation was also incorrectly listing set operations (EXCEPT/UNION etc) as operators of SELECT in particular, whereas they actually work also with any selection clause.

The root cause of this sorry state of affairs was caused by a single
misconception: there is really no such thing as a "SELECT statement"
in the SQL grammar; instead there are various relational expressions
and operators, and selection clauses are just a particular instance of relational expressions.

This patch enhances this situation as follows:

- The page about a "SELECT statement" is deleted. It is replaced by a
  page titled "Simple SELECT clause", to properly highlight it is
  merely an instance of a selection clause (albeit with a reminder
  that it happens to also be valid as standalone statement).

- To ensure the basic SELECT syntax remains the "main SELECT doc page"
  for newcomers, the top-level links in navigation for the `SELECT`
  keyword are changed to point to the page on the simple SELECT
  clause.

- A new page "Relational Expressions" is introduced, which explains
  properly the concept of "Statement-like Query". The pages about
  other statements that require a statement-like query are modified to
  point to this new page.

- The difference between "table expressions", "selection clauses" and
  "statement-like queries" is repeatedly clarified in numerous places,
  to ensure that clarity is pervasively maintained throughout the
  docs.

Additionally, the syntax diagrams for all these constructs are updated to more accurately reflect reality. Some more documentation for the various `JOIN` operators is added.

To review this patch, the reviewer is invited to first proceed in a
holistic manner and navigate the result of the changes with the eyes
of a newcomer:

1. Land on the "Simple SELECT clause" page; ensure that it makes
   sense, including its simplified syntax diagram.

2. Check out the page on "Table Expressions", which documents what can
   fit in the FROM clause of a simple SELECT clause. Check that it
   makes sense, including the new syntax diagrams.

2. Check the docs for INSERT/UPDATE/UPSERT/DELETE and check they make
   sense, including the new syntax diagrams.

3. Navigate to the page on "Relational Expressions" and check that it
   makes sense, including the new syntax diagrams.

4. Review the new page "Limiting Query Results" linked from there.

Once the reviewer satisfies themselves that the relationship between
the pages and the syntax diagrams make sense, the line-by-line diff
can be reviewed for further edits.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/docs/2762)
<!-- Reviewable:end -->

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/7a7bd42fdac1b0b522c7968a14063611fdd29ef3/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/f05055a49de66f6de51aa6e9b0c64cac95595765/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/a31ffdfb3c0c80500a47fb02f8bd2e74e454ef54/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/1259d3fbcb1d991c17a2aa2ab00a7e6717b351ff/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/bc2ca99ad30708948307de418d67d2388c0083d2/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/f23efd1d9e4bc9d10d2c659460962c388e1f315c/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/be80969f6bf3281eda9b73a1546751d1706ac951/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/27213f9ed95bd96dee55cab0b19fcaa1ece6e0ff/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/96adf10b6722936c8e108d178ae6489018009aa2/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/790a73da4a738ec5ab5c3de7c6f4a779fc6a0d6a/

knz commented:
@jseldess please check the PR description for suggestions on how to review this. I suggest we spend our scheduled time together later this week to do a live review.

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/6d11015f38718f3e21813b4859e63fe57792fb22/

jseldess commented:
Thank you, @knz! I've only had a chance to skim, but this looks like a significant improvement. I'll read through carefully tomorrow.

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/3b039febef857fcec8c2bb80fa50b90bb91e0d12/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0c76fbaa3b8ac589d059b2e582f854ab2d8da4f6/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/18ad8a46b767196ef704c5965668a5fc9ecd2f0a/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/598db6d2744a4d05e67aff2622ed7748c04362d3/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/efd5688f6636950a5e263060a9fb313a9f61e388/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/094b0aa8c0057ef7c2e018acc5a041b63de09494/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/57717db8c9534098c7b299e40c722adc6faaafbc/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0f75ae1034a17853705ada21137566db2c2a517e/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/047370c4ef550254b5fa755f313a452eee47e073/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/a3385f3ee51d5e97f66664dc717ee2989e890608/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/1319f4f8f19b6da20a919987883a971ea0edef3e/

jseldess commented on v2.0/select-clause.md:
```diff
@@ -0,0 +1,451 @@
+| `table_ref` | The [table expression](table-expressions.html) you want to retrieve data from.|
+| `AS OF SYSTEM TIME timestamp` | Retrieve data as it existed as of [`timestamp`](timestamp.html).<br/><br/>For more information, see [this example](#select-historical-data-time-travel) or our blog post [Time-Travel Queries](https://www.cockroachlabs.com/blog/time-travel-queries-select-witty_subtitle-the_future/).<br/><br/>Because `AS OF SYSTEM TIME` returns historical data, your reads might be s ...
+| `WHERE a_expr` | Only retrieve rows that return `TRUE` for `a_expr`, which must be a [scalar expression](scalar-expressions.html) that returns Boolean values using columns (e.g., `<column> = <value>`).  |
+| `GROUP BY expr_list` | When using [aggregate functions](functions-and-operators.html#aggregate-functions) in `target_elem` or `HAVING`, list the column groupings in `expr_list`. |
```
In the diagram, it looks like we inlined `expr_list`, so we should either adjust this row or the diagram.

jseldess commented on v2.0/select-clause.md:
```diff
@@ -0,0 +1,451 @@
+| `HAVING a_expr` | Only retrieve aggregate function groups that return `TRUE` for `a_expr`, which must be a [scalar expression](scalar-expressions.html) that returns Boolean values using an aggregate function (e.g., `<aggregate function> = <value>`). <br/><br/>`HAVING` works like the `WHERE` clause, but for aggregate functions. |
+| `WINDOW window_definition_list` | A list of [window functions definitions](window-functions.html) |
+
+{{site.data.alerts.callout_info}}Using two or more table expressions in the <code>FROM</code> sub-clause,
```
This is very helpful, but I think it should live in the `table_ref` description, though you won't be able to use the callout styling. I'd suggest:

```
The [table expression](table-expressions.html) you want to retrieve data from.<br><br>Using two or more table expressions in the `FROM` sub-clause, separated with a comma, is equivalent to a [`CROSS JOIN`](joins.html) expression.
```

jseldess commented on v2.0/select-clause.md:
```diff
@@ -0,0 +1,451 @@
+
+#### Retrieve Specific Columns
+
+Retrieve specific columns by naming them in a comma-separated list.
```
End with a colon and place `{% include copy-clipboard.html %}` on line before code block.

jseldess commented on v2.0/select-clause.md:
```diff
@@ -0,0 +1,451 @@
+
+#### Retrieve All Columns
+
+Retrieve all columns by using `*`.
```
End with a colon and place `{% include copy-clipboard.html %}` on line before code block.

jseldess commented on v2.0/select-clause.md:
```diff
@@ -0,0 +1,451 @@
+
+#### Filter on a Single Condition
+
+Filter rows with expressions that use columns and return Boolean values in the `WHERE` clause.
```
End with a colon and place `{% include copy-clipboard.html %}` on line before code block.

jseldess replied:
Same applies to all other examples, but some of the intro sentences may need to be tweaked a bit to accommodate that form. 

I'm happy to make these edits for you, if you like.

jseldess commented on v2.0/select-clause.md:
```diff
@@ -0,0 +1,451 @@
+| `WHERE a_expr` | Only retrieve rows that return `TRUE` for `a_expr`, which must be a [scalar expression](scalar-expressions.html) that returns Boolean values using columns (e.g., `<column> = <value>`).  |
+| `GROUP BY expr_list` | When using [aggregate functions](functions-and-operators.html#aggregate-functions) in `target_elem` or `HAVING`, list the column groupings in `expr_list`. |
+| `HAVING a_expr` | Only retrieve aggregate function groups that return `TRUE` for `a_expr`, which must be a [scalar expression](scalar-expressions.html) that returns Boolean values using an aggregate function (e.g., `<aggregate function> = <value>`). <br/><br/>`HAVING` works like the `WHERE` clause, but for aggregate functions. |
+| `WINDOW window_definition_list` | A list of [window functions definitions](window-functions.html) |
```
Add period at end of sentence.

jseldess commented on v2.0/select-clause.md:
```diff
@@ -0,0 +1,451 @@
+title: Simple SELECT Clause
+summary: The Simple SELECT clause loads or computes data from various sources.
+toc: false
+---
```
Since we're renaming this file, we need to add a client-side redirect to the front-matter:

```
redirect-from: select.html
```

jseldess commented on v2.0/subqueries.md:
```diff
@@ -0,0 +1,59 @@
+# Find every customer with at least one order.
+> SELECT c.name
+    FROM customers c
+   WHERE EXISTS(SELECT * FROM orders o WHERE o.customer_id = c.id);
```
nit: Indent to match `FROM`.

jseldess commented on v2.0/subqueries.md:
nit: Indent to match `FROM`.

jseldess commented on v2.0/table-expressions.md:
```diff
@@ -5,44 +5,56 @@ toc: false
+## Table Expressions Language
 
-CockroachDB recognizes the following table expressions:
+The synopsis above really defines a mini-language to construct
```
`more simple` > `simpler`.

jseldess commented on v2.0/table-expressions.md:
```diff
@@ -141,55 +207,101 @@ For example:
++---------+------------+
+~~~
+
+{{site.data.alerts.callout_info}}<code>WITH ORDINALITY</code> necessarily prevents some optimizations of the surrounding query. Use it sparingly if performance is a concern, and always check the output of <a href="explain.html">EXPLAIN</a> in case of doubt. {{site.data.alerts.end}}
```
Use `<code>` for `EXPLAIN` as well.

jseldess commented on v2.0/table-expressions.md:
```diff
@@ -141,55 +207,101 @@ For example:
+{{site.data.alerts.callout_info}}See also <a href="subqueries.html">Subqueries</a> for more details and performance best practices.{{site.data.alerts.end}}
 
-Currently works only with small data sets; find more info in our [blog post](https://www.cockroachlabs.com/blog/cockroachdbs-first-join/).
+{{site.data.alerts.callout_info}}To use other statements that produce data in a table expression, for example `SHOW`, use the <a href="#using-the-output-of-other-statements">square bracket notation</a>.{{site.data.alerts.end}}
```
I'd just include this in the previous callout, as a separate paragraph. 

Also, you can't use backticks inside a callout, so you need to use `<code>SHOW</code>`.

jseldess commented on v2.0/update.md:
```diff
@@ -16,23 +16,26 @@ The user must have the `SELECT` and `UPDATE` [privileges](privileges.html) on th
-`a_expr` | The new value you want to use, the [aggregate function](functions-and-operators.html#aggregate-functions) you want to perform, or the [value expression](sql-expressions.html) you want to use.
+`a_expr` | The new value you want to use, the [aggregate function](functions-and-operators.html#aggregate-functions) you want to perform, or the [scalar expression](scalar-expressions.html) you want to use.
 `DEFAULT` | To fill columns with their [default values](default-value.html), use `DEFAULT VALUES` in place of `a_expr`. To fill a specific column with its default value, leave the value out of the `a_expr` or use `DEFAULT` at the appropriate position.
 `column_name_list` | A comma-separated list of column names, in parentheses.
```
We inlined this in the diagram, so there's a mismatch. We could either not inline in the diagram, or update this row.

jseldess commented on v2.0/insert.md:
```diff
@@ -34,14 +34,41 @@ table td:first-child {
 Parameter | Description
 ----------|------------
-`opt_with_clause` | Docs coming soon.
+`common_table_expr` | Docs coming soon.
```
Asked this elsewhere, but do you plan to document this as part of this PR?

jseldess commented:
Done reviewing for now, @knz. Thanks again! I think this PR also closes https://github.com/cockroachdb/docs/issues/2719 and https://github.com/cockroachdb/docs/issues/788

knz commented:
Hi Jesse,
thank you for the quality comments. I will process them soon.

In the meantime, I have added the missing page on common table expressions. Perhaps @jordanlewis you want to check I did them justice?

Jesse, I think this PR indeed improves upon #788 but IMHO it does not fully resolve it. We can chat about it.

---

Review status: 0 of 66 files reviewed at latest revision, 27 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-:-L8iav3u68jLb1TFFGA8:b-dfbjsx)*
<!-- Sent from Reviewable.io -->

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/eaf8960577be2a571d85d4911640a9072f245cfc/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/844028e912ce8fd2a1492c70e4babab009f569ac/

knz commented:
I  have addressed most of the remaining comments. PTAL.

---

Review status: 0 of 68 files reviewed at latest revision, 27 unresolved discussions, some commit checks pending.

---

*[v2.0/insert.md, line 37 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L89glgu1roQ5FYR27LU-r1-37:-L8igHNs58uTVPowkQvQ:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/insert.md#L37)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Asked this elsewhere, but do you plan to document this as part of this PR?
</blockquote></details>

Done.

---

*[v2.0/joins.md, line 10 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8XOBaNtEBg-ndp6WbN-r1-10:-L8igLvFBlSniLaNe16q:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/joins.md#L10)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

`a particular kind of table expressions` > `a particular kind of table expression`
</blockquote></details>

Done.

---

*[v2.0/joins.md, line 25 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8XOBaNtEBg-ndp6WbN-r1-25:-L8igO1-29i9XeINAeX8:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/joins.md#L25)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

`Expression` > `expression`
</blockquote></details>

Done.

---

*[v2.0/relational-expressions.md, line 5 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8TpgPH--CfUHlc7XsZ-r1-5:-L8igUYF5v3eyZJ6TuXo:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/relational-expressions.md#L5)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Since we're removing `sql-expressions.md`, we need to put a client-side redirect in the front-matter here:

```
redirect-from: sql-expressions.html
```
</blockquote></details>

Done.

---

*[v2.0/relational-expressions.md, line 47 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8TpgPH--CfUHlc7XsZ-r1-47:-L8igXp7AdYj-0GQVPaJ:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/relational-expressions.md#L47)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Did you want to cover this feature as part of this PR?
</blockquote></details>

Done.

---

*[v2.0/select-clause.md, line 5 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8TpgPH--CfUHlc7Xs_-r1-5:-L8igZhR-d3l3k6bvt_w:b-muo11a) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/select-clause.md#L5)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Since we're renaming this file, we need to add a client-side redirect to the front-matter:

```
redirect-from: select.html
```
</blockquote></details>

The select page still exists. A redirect would be incorrect I think.

---

*[v2.0/select-clause.md, line 35 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8TpgPH--CfUHlc7Xs_-r1-35:-L8igkvH4OWlGT4ZLpkQ:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/select-clause.md#L35)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

`[Eliminate Duplicate Rows]` > `[Eliminate Duplicate Rows](#eliminate-duplicate-rows)`
</blockquote></details>

Done.

---

*[v2.0/select-clause.md, line 36 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8TpgPH--CfUHlc7Xs_-r1-36:-L8igpYB4x7U7uKwsCyG:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/select-clause.md#L36)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

I'd remove `to use`.
</blockquote></details>

Done.

---

*[v2.0/select-clause.md, line 40 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8TpgPH--CfUHlc7Xs_-r1-40:-L8ih5dc4Q9dOG8anmye:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/select-clause.md#L40)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

In the diagram, it looks like we inlined `expr_list`, so we should either adjust this row or the diagram.
</blockquote></details>

Done.

---

*[v2.0/select-clause.md, line 42 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8TpgPH--CfUHlc7Xs_-r1-42:-L8ih7R131kUflTBEUyg:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/select-clause.md#L42)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Add period at end of sentence.
</blockquote></details>

Done.

---

*[v2.0/select-clause.md, line 44 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8TpgPH--CfUHlc7Xs_-r1-44:-L8ihKYFDVrLL6L6Yg6z:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/select-clause.md#L44)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

This is very helpful, but I think it should live in the `table_ref` description, though you won't be able to use the callout styling. I'd suggest:

```
The [table expression](table-expressions.html) you want to retrieve data from.<br><br>Using two or more table expressions in the `FROM` sub-clause, separated with a comma, is equivalent to a [`CROSS JOIN`](joins.html) expression.
```
</blockquote></details>

Done.

---

*[v2.0/select-clause.md, line 67 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8TpgPH--CfUHlc7Xs_-r1-67:-L8ihmKqCs_x-KWJtpdO:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/select-clause.md#L67)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

End with a colon and place `{% include copy-clipboard.html %}` on line before code block.
</blockquote></details>

Done.

---

*[v2.0/select-clause.md, line 86 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8TpgPH--CfUHlc7Xs_-r1-86:-L8ihnLk1M7jR5KRh-aM:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/select-clause.md#L86)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

End with a colon and place `{% include copy-clipboard.html %}` on line before code block.
</blockquote></details>

Done.

---

*[v2.0/select-clause.md, line 107 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8TpgPH--CfUHlc7Xs_-r1-107:-L8ihoHs7ZDQjLiWSwxc:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/select-clause.md#L107)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Same applies to all other examples, but some of the intro sentences may need to be tweaked a bit to accommodate that form. 

I'm happy to make these edits for you, if you like.
</blockquote></details>

Done.

---

*[v2.0/selection-clauses.md, line 36 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L89rh83oIxB2ECSBSoM-r1-36:-L8iiLSi71OkguD21hDp:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/selection-clauses.md#L36)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Also, I think we want `SELECT` to link to the relevant section on this page, so we should change `<a href="select-clause.html">` to `<a href="#select-clause">`.
</blockquote></details>

Done.

---

*[v2.0/selection-clauses.md, line 114 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L89rh83oIxB2ECSBSoM-r1-114:-L8ii7ns7U8p99oBX9kL:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/selection-clauses.md#L114)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

nit: `standalone` > `stand-alone`
</blockquote></details>

Done.

---

*[v2.0/selection-clauses.md, line 116 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L89rh83oIxB2ECSBSoM-r1-116:-L8iiT4g0N4BYI7LDBRb:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/selection-clauses.md#L116)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

`the other two`? Should this be `the other three`?
</blockquote></details>

Done.

---

*[v2.0/set-operations.md, line 7 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L89rh83oIxB2ECSBSoN-r1-7:-L8iiaDo8Sg6N_Y1Gqcx:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/set-operations.md#L7)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

nit: I'd remove the article and fix the verb, e.g., `Set operations combine data...`
</blockquote></details>

Done.

---

*[v2.0/set-operations.md, line 32 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L89rh83oIxB2ECSBSoN-r1-32:-L8iiar70x2SY7TZxZra:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/set-operations.md#L32)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Let's add the copy-to-clipboard feature to all statement blocks.
</blockquote></details>

Done.

---

*[v2.0/subqueries.md, line 29 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8XXTdAVuayPIm3xNru-r1-29:-L8ijbNg3Od74ZP--Jwm:b-oh3ezv) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/subqueries.md#L29)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

nit: Indent to match `FROM`.
</blockquote></details>

I don't understand?

---

*[v2.0/subqueries.md, line 42 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8XXTdAVuayPIm3xNru-r1-42:-L8iivtRAi5b1l-HvM7E:b-oh3ezv) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/subqueries.md#L42)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

nit: Indent to match `FROM`.
</blockquote></details>

I don't understand?

---

*[v2.0/table-expressions.md, line 35 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8TpgPH--CfUHlc7XsW-r1-35:-L8ijBR38vKLbkJub4d0:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/table-expressions.md#L35)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

`more simple` > `simpler`.
</blockquote></details>

Done.

---

*[v2.0/table-expressions.md, line 249 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8TpgPH--CfUHlc7XsW-r1-249:-L8ijEHB4xTj6nMwxavN:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/table-expressions.md#L249)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Use `<code>` for `EXPLAIN` as well.
</blockquote></details>

Done.

---

*[v2.0/table-expressions.md, line 287 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8TpgPH--CfUHlc7XsW-r1-287:-L8ijO4Z0zc5vom5qicb:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/table-expressions.md#L287)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

I'd just include this in the previous callout, as a separate paragraph. 

Also, you can't use backticks inside a callout, so you need to use `<code>SHOW</code>`.
</blockquote></details>

Done.

---

*[v2.0/update.md, line 33 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L89jnyKMjvXFZgJ-Qgy-r1-33:-L8ijYQ33bAlhuX94XA4:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/update.md#L33)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

We inlined this in the diagram, so there's a mismatch. We could either not inline in the diagram, or update this row.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-:-L8ijsPk313wJayhblB7:b5nlz8z)*
<!-- Sent from Reviewable.io -->

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/082d82b33c5adedb1ad4766bc1b8cbb0acd0ae3e/

knz commented:
Review status: 0 of 68 files reviewed at latest revision, 27 unresolved discussions, some commit checks failed.

---

*[v2.0/relational-expressions.md, line 5 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8TpgPH--CfUHlc7XsZ-r1-5:-L8ik7Mo58vD21kUGR-8:b-6xt0xc) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/relational-expressions.md#L5)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

(the new page is scalar-expressions.md, not this one)

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/docs/2762)*
<!-- Sent from Reviewable.io -->

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/88fb99f07efd321890f16b2fc08ae68db77f3d7d/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/dfae26381c0db5feadbbb2ef38f9a500ff8d65fa/

jordanlewis commented:
@knz this is really excellent work! I have reviewed the new page on CTEs - I think it looks fantastic. Thank you!

jseldess commented:
Review status: 0 of 68 files reviewed at latest revision, 9 unresolved discussions, all commit checks successful.

---

*[v2.0/relational-expressions.md, line 5 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8TpgPH--CfUHlc7XsZ-r1-5:-L8jeDp72TUnDb5DpVvQ:b-1zhu54) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/relational-expressions.md#L5)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

(the new page is scalar-expressions.md, not this one)
</blockquote></details>

Ah, ok, thank you!

---

*[v2.0/select-clause.md, line 5 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8TpgPH--CfUHlc7Xs_-r1-5:-L8jes8TC00SOH9X3sTv:b2ggd6b) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/select-clause.md#L5)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

The select page still exists. A redirect would be incorrect I think.
</blockquote></details>

Ah, I see now. But do we really need to keep the `select.md` page around? Since we link to the simple select clause page pretty much everywhere, it's probably ok to just remove `select.md` and add `redirect-from: select.html` to the front-matter of this new page. Thoughts?

---

*[v2.0/select-clause.md, line 86 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8TpgPH--CfUHlc7Xs_-r1-86:-L8jfaFIAVic4R1o3VcJ:b-lyn5nh) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/select-clause.md#L86)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Thanks, but you could still replace the period with a colon. Also true in many of the other sentences introducing an example. 

---

*[v2.0/select-clause.md, line 107 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8TpgPH--CfUHlc7Xs_-r1-107:-L8jflLI0NeEaPnDU7n5:br0a87z) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/select-clause.md#L107)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

As mentioned above, you could still use a colon to introduce each example.

---

*[v2.0/selection-clauses.md, line 114 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L89rh83oIxB2ECSBSoM-r1-114:-L8jgAGWB84uky64FcAS:b-argfoq) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/selection-clauses.md#L114)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Would you mind searching for `standalone` in your other files as well and making the same change?

---

*[v2.0/subqueries.md, line 29 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8XXTdAVuayPIm3xNru-r1-29:-L8jg_Tx1wEwm6QjMvuF:b-21zslb) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/subqueries.md#L29)):*
Sorry, that wasn't clear. The indentation of the clauses is strange. I'd probably just left-align all but the first line, as you've done in other examples. For example:

```
> SELECT c.name
FROM customers c
WHERE EXISTS(SELECT * FROM orders o WHERE o.customer_id = c.id);
``` 

---

*[v2.0/subqueries.md, line 42 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8XXTdAVuayPIm3xNru-r1-42:-L8jgiOKB6XOpTZZCnEx:bbx87r0) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/subqueries.md#L42)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

I don't understand?
</blockquote></details>

See comment above. 

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/docs/2762)*
<!-- Sent from Reviewable.io -->

jseldess commented:
Review status: 0 of 68 files reviewed at latest revision, 9 unresolved discussions, all commit checks successful.

---

*[v2.0/common-table-expressions.md, line 2 at r2](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8ji7cFAQ307KfInpMw:-L8ji7cFAQ307KfInpMx:bwx32tc) ([raw file](https://github.com/cockroachdb/docs/blob/dfae26381c0db5feadbbb2ef38f9a500ff8d65fa/v2.0/common-table-expressions.md#L2)):*
> ```Markdown
> ---
> title: Common Table Expressions
> ```

This page is great! Thank you, @knz. 

We still need to add it to the sidenav. Any ideas where it belongs? Also, would you mind updating the `Common Table Expressions` row in `sql-feature-support.md`?

---

*[v2.0/common-table-expressions.md, line 22 at r2](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8jiTLp68WkhX7ki9SY:-L8jiTLp68WkhX7ki9SZ:bmvnmch) ([raw file](https://github.com/cockroachdb/docs/blob/dfae26381c0db5feadbbb2ef38f9a500ff8d65fa/v2.0/common-table-expressions.md#L22)):*
> ```Markdown
> ## Synopsis
> 
> {% include sql/{{ page.version.version }}/diagrams/with_clause.html %}
> ```

The markup following this include is broken, which someone seems to happen. The (hacky) solution is to wrap this in `<section>` tags:

```
<section>% include sql/{{ page.version.version }}/diagrams/with_clause.html %}</section>
```

---

*[v2.0/common-table-expressions.md, line 41 at r2](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8jjCiI9gu7npwPcpSG:-L8jjCiI9gu7npwPcpSH:b-bwjp3d) ([raw file](https://github.com/cockroachdb/docs/blob/dfae26381c0db5feadbbb2ef38f9a500ff8d65fa/v2.0/common-table-expressions.md#L41)):*
> ```Markdown
> 
> {% include copy-clipboard.html %}
> ~~~ sql
> ```

I'd left-align all but the first line, as you've done in most other examples.

---

*[v2.0/common-table-expressions.md, line 56 at r2](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8jjNAWB8nsarlhnW_v:-L8jjNAWB8nsarlhnW_w:bv5ghjo) ([raw file](https://github.com/cockroachdb/docs/blob/dfae26381c0db5feadbbb2ef38f9a500ff8d65fa/v2.0/common-table-expressions.md#L56)):*
> ```Markdown
> 
> {% include copy-clipboard.html %}
> ~~~ sql
> ```

Same as above. But perhaps you've formatted it this way intentionally and I just don't understand? Feel free to disregard.

---

*[v2.0/common-table-expressions.md, line 134 at r2](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8jkjbN11KVgoq46dWU:-L8jkjbN11KVgoq46dWV:b-tlfjdg) ([raw file](https://github.com/cockroachdb/docs/blob/dfae26381c0db5feadbbb2ef38f9a500ff8d65fa/v2.0/common-table-expressions.md#L134)):*
> ```Markdown
> a data-modifying statement (<code>INSERT</code>, <code>DELETE</code>,
> etc.), the modifications are performed fully even if only part
> of the results are used, e.g. with <a
> ```

nit: `e.g.` > `e.g.,`

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/docs/2762)*
<!-- Sent from Reviewable.io -->

knz commented:
Review status: 0 of 70 files reviewed at latest revision, 14 unresolved discussions.

---

*[v2.0/common-table-expressions.md, line 2 at r2](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8ji7cFAQ307KfInpMw:-L8ldoXG3_EnMobwN0f_:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/dfae26381c0db5feadbbb2ef38f9a500ff8d65fa/v2.0/common-table-expressions.md#L2)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

This page is great! Thank you, @knz. 

We still need to add it to the sidenav. Any ideas where it belongs? Also, would you mind updating the `Common Table Expressions` row in `sql-feature-support.md`?
</blockquote></details>

Done.

---

*[v2.0/common-table-expressions.md, line 22 at r2](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8jiTLp68WkhX7ki9SY:-L8ldthx9NfYKnrn5fEc:bx837l) ([raw file](https://github.com/cockroachdb/docs/blob/dfae26381c0db5feadbbb2ef38f9a500ff8d65fa/v2.0/common-table-expressions.md#L22)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

The markup following this include is broken, which someone seems to happen. The (hacky) solution is to wrap this in `<section>` tags:

```
<section>% include sql/{{ page.version.version }}/diagrams/with_clause.html %}</section>
```
</blockquote></details>

As I explained to Rich earlier, `<section>` is not really appropriate: it has semantic value (create a section in the doc) and this semantic value is not appropriate for just a diagram.
Used `<div>` instead.

---

*[v2.0/common-table-expressions.md, line 56 at r2](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8jjNAWB8nsarlhnW_v:-L8lbKj0963L6hQ8mUus:b-ygivpz) ([raw file](https://github.com/cockroachdb/docs/blob/dfae26381c0db5feadbbb2ef38f9a500ff8d65fa/v2.0/common-table-expressions.md#L56)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Same as above. But perhaps you've formatted it this way intentionally and I just don't understand? Feel free to disregard.
</blockquote></details>

I have learned this style from some books, but you can see some example online:
http://sqlite.org/sqllogictest/artifact/8bf320a7c5bd7456

It makes the queries much easier to read, by ensuring that the important part of the queries (the operands to various clauses, and thus not the keywords) are all aligned on the same column.

---

*[v2.0/common-table-expressions.md, line 134 at r2](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8jkjbN11KVgoq46dWU:-L8lbsjU-AdgQHUGtLes:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/dfae26381c0db5feadbbb2ef38f9a500ff8d65fa/v2.0/common-table-expressions.md#L134)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

nit: `e.g.` > `e.g.,`
</blockquote></details>

Done.

---

*[v2.0/select-clause.md, line 5 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8TpgPH--CfUHlc7Xs_-r1-5:-L8lbyPI2aJaa9pcwPF0:b2zrp5z) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/select-clause.md#L5)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Ah, I see now. But do we really need to keep the `select.md` page around? Since we link to the simple select clause page pretty much everywhere, it's probably ok to just remove `select.md` and add `redirect-from: select.html` to the front-matter of this new page. Thoughts?
</blockquote></details>

I'm ok with that. Done.

---

*[v2.0/select-clause.md, line 86 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8TpgPH--CfUHlc7Xs_-r1-86:-L8lcw0t52W6mlhFrpC5:bo86ytm) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/select-clause.md#L86)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Thanks, but you could still replace the period with a colon. Also true in many of the other sentences introducing an example. 
</blockquote></details>

Done, but I'd like to highlight I merely copied this over from select.md.

---

*[v2.0/select-clause.md, line 107 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8TpgPH--CfUHlc7Xs_-r1-107:-L8ld-Bj9cCbdCiqdoNi:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/select-clause.md#L107)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

As mentioned above, you could still use a colon to introduce each example.
</blockquote></details>

Done.

---

*[v2.0/selection-clauses.md, line 114 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L89rh83oIxB2ECSBSoM-r1-114:-L8ldBqM-Xi4kDHjLzY7:bf82ioc) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/selection-clauses.md#L114)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Would you mind searching for `standalone` in your other files as well and making the same change?
</blockquote></details>

There are several occurrences out of this PR. I changed them as well.

I am not totally fond of this change however. I have never seen this spelled this way.

---

*[v2.0/subqueries.md, line 29 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8XXTdAVuayPIm3xNru-r1-29:-L8ldS621CxaEG3ghhJf:b25xwdd) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/subqueries.md#L29)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Sorry, that wasn't clear. The indentation of the clauses is strange. I'd probably just left-align all but the first line, as you've done in other examples. For example:

```
> SELECT c.name
FROM customers c
WHERE EXISTS(SELECT * FROM orders o WHERE o.customer_id = c.id);
``` 
</blockquote></details>

See my other comment.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/docs/2762)*
<!-- Sent from Reviewable.io -->

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/79116e0bcb098f0f0656fb1c8218d779de0b5d7d/

rmloveland commented:
Review status: 0 of 70 files reviewed at latest revision, 14 unresolved discussions, all commit checks successful.

---

*[v2.0/common-table-expressions.md, line 22 at r2](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8jiTLp68WkhX7ki9SY:-L8mANo1-5E08e1RYNXV:b-xys0q7) ([raw file](https://github.com/cockroachdb/docs/blob/dfae26381c0db5feadbbb2ef38f9a500ff8d65fa/v2.0/common-table-expressions.md#L22)):*
> As I explained to Rich earlier, <section> is not really appropriate: it has semantic value

The docs [wiki page on these diagrams](https://github.com/cockroachdb/docs/wiki/SQL-Grammar-Railroad-Diagram-Changes) was recommending the use of `<section>`, I just updated it to recommend `<div>`.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/docs/2762)*
<!-- Sent from Reviewable.io -->

jseldess commented:
Review status: 0 of 70 files reviewed at latest revision, 8 unresolved discussions, all commit checks successful.

---

*[v2.0/common-table-expressions.md, line 22 at r2](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8jiTLp68WkhX7ki9SY:-L8mDuBxDsLpyS2ccpGK:b-77cgbt) ([raw file](https://github.com/cockroachdb/docs/blob/dfae26381c0db5feadbbb2ef38f9a500ff8d65fa/v2.0/common-table-expressions.md#L22)):*
<details><summary><i>Previously, rmloveland (Rich Loveland) wrote…</i></summary><blockquote>

> As I explained to Rich earlier, <section> is not really appropriate: it has semantic value

The docs [wiki page on these diagrams](https://github.com/cockroachdb/docs/wiki/SQL-Grammar-Railroad-Diagram-Changes) was recommending the use of `<section>`, I just updated it to recommend `<div>`.
</blockquote></details>

Thanks, @knz and @rmloveland. There are some strange edge cases where we've only been able to get the markdown to convert properly by using `<section>` instead of `<div>`. This has come up here, for example: https://github.com/cockroachdb/docs/issues/2146#issuecomment-369931123. In any case, it does sound best to recommend `<div>`. When we run into trouble, we can look more closely.

---

*[v2.0/common-table-expressions.md, line 56 at r2](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8jjNAWB8nsarlhnW_v:-L8mEO8hAdJlc1GSwkRE:bne4ax6) ([raw file](https://github.com/cockroachdb/docs/blob/dfae26381c0db5feadbbb2ef38f9a500ff8d65fa/v2.0/common-table-expressions.md#L56)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

I have learned this style from some books, but you can see some example online:
http://sqlite.org/sqllogictest/artifact/8bf320a7c5bd7456

It makes the queries much easier to read, by ensuring that the important part of the queries (the operands to various clauses, and thus not the keywords) are all aligned on the same column.
</blockquote></details>

Thanks for the context. We should definitely include conventions around multi-line sql samples in the style guide that @lhirata will be driving in the 2.1 release cycle. Lauren, please use Raphael as a resource for that. 

---

*[v2.0/relational-expressions.md, line 13 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8TpgPH--CfUHlc7XsZ-r1-13:-L8mEt6S7VfmRCyI8Tu_:btlwvth) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/relational-expressions.md#L13)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Instead of this overview section, I think we should add the other flavors of relational expressions as sections on this page, providing the high-level info you have here with links away to the more details docs. 
</blockquote></details>

@knz, you didn't respond this comment, so if there's time, we can discuss in our meeting today. This is a nit, really, and not something that needs to be done now.

---

*[v2.0/select-clause.md, line 86 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8TpgPH--CfUHlc7Xs_-r1-86:-L8mF45B1vR_KX9R-EMr:bgfy52n) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/select-clause.md#L86)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

Done, but I'd like to highlight I merely copied this over from select.md.
</blockquote></details>

Thanks, @knz, and apologies for the current inconsistencies across our docs. As mentioned elsewhere, we're going to work out more thorough style guidelines/conventions in the 2.1 cycle.

---

*[v2.0/selection-clauses.md, line 114 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L89rh83oIxB2ECSBSoM-r1-114:-L8mFR1k-hqoOQcHLmGe:b-fke242) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/selection-clauses.md#L114)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

There are several occurrences out of this PR. I changed them as well.

I am not totally fond of this change however. I have never seen this spelled this way.
</blockquote></details>

Yeah, this is pure preference. Both forms are correct, though the hyphenated form is more widely used, I think. 

---

*[v2.0/subqueries.md, line 29 at r1](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L8XXTdAVuayPIm3xNru-r1-29:-L8mFeIBDVWSsBzHSJCw:b-zv596) ([raw file](https://github.com/cockroachdb/docs/blob/1319f4f8f19b6da20a919987883a971ea0edef3e/v2.0/subqueries.md#L29)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

See my other comment.
</blockquote></details>

In a lot of cases, you have just left-aligned all lines but the first, so I'm not really clear on when you do that vs. indent. But this is not worth a lot of time right now.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/docs/2762)*
<!-- Sent from Reviewable.io -->

jseldess commented on v2.0/select-clause.md:
```diff
@@ -1,52 +1,70 @@
+title: Simple SELECT Clause
+summary: The Simple SELECT clause loads or computes data from various sources.
 toc: false
+redirect-from: select.html
```
One more field to add to the front-matter to make our version switcher work between 1.1 and 2.0:

```
key: select.html
```

jseldess commented on v2.0/scalar-expressions.md:
```diff
@@ -1,17 +1,18 @@
+summary: Scalar expressions allow the computation of new values from basic parts.
 toc: false
+redirect-from: sql-expressions.html
 ---
```
One more field to add to get our version selector to work between 1.1 and 2.0:

```
key: sql-expressions.html
```

jseldess commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>!!!!!! with a few minor edits left.

---







*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-:-L8mHddH8RVg8iqsqNyB:bgcums3)*
<!-- Sent from Reviewable.io -->

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/4efd6caf4be17bb51a31cbf63dd00dcdd9e36319/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/5e99890c23c16d2c0ac0ffb2ab8b24f53de53fb2/

jseldess commented:
Thank you for iterating, @knz. I'm now concerned about the proliferation of terms we use to refer to the same thing: `stand-alone queries`, `statement-like queries`, `statement-like relational expressions`, and possibly others. These terms are not intuitive in themselves, and don't seem to appear in the mysql or postgres docs, so if we're going to use a term that is not already conventional, we need to choose 1 and stick with it. Can you please make that change? I'd probably suggest "statement-like queries" for now. 

More broadly, on further review and consideration, I have to admit that I'm still confused and a bit concerned about this level of abstraction. I *think*, if we wanted to keep things simpler, we could get away with covering these semantics just on the selection clauses page, given the definition of statement-like queries as "a group one or more selection clause with set operations". But it is late in the game to request such a change, and we do need to merge soon, so I'm ok holding off and looping other people in for perspective post-2.0 release. 

---







*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-:-L8ymbbK6xfS0sWlhaLx:b5ux2u0)*
<!-- Sent from Reviewable.io -->

knz commented:
Jesse, thanks for the feedback. I agree with it and I am glad that we could get other more superficial issues out of the way earlier in the review process to reveal the more important issues.

I have decided to collapse the pages on set operations, selection clauses and relational expressions into a single page "selection queries". I think this term is both suitably general, simple and accurate.

Let me know what you think.

---

Review status: 0 of 71 files reviewed at latest revision, 10 unresolved discussions.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-:-L91Kozp1UNerJSvd1Bt:b1f8rze)*
<!-- Sent from Reviewable.io -->

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/2d6815c430dd08a773ca2fd5253b282491a2abf8/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/16b5ff44f3d0bcef22751b34786a9f511e3c316a/

jseldess commented:
This LGTM! Just a few final nits. Thank you for putting in so much time and effort.

---

Review status: 0 of 70 files reviewed at latest revision, 10 unresolved discussions, all commit checks successful.

---

*[v2.0/selection-queries.md, line 11 at r4](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L91YJXQEn5Hw2_cLjwJ:-L91YJXQEn5Hw2_cLjwK:b-msk8al) ([raw file](https://github.com/cockroachdb/docs/blob/16b5ff44f3d0bcef22751b34786a9f511e3c316a/v2.0/selection-queries.md#L11)):*
> ```Markdown
> Selection queries read and process data in CockroachDB.  They are more
> general than [simple `SELECT` clauses](select-clause.html): they can
> group one or more [selection clause](#selection-clauses) with [set
> ```

one or more selection clauses

---

*[v2.0/selection-queries.md, line 15 at r4](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L91YWciDnrignWZz-fJ:-L91YWciDnrignWZz-fK:b-i848so) ([raw file](https://github.com/cockroachdb/docs/blob/16b5ff44f3d0bcef22751b34786a9f511e3c316a/v2.0/selection-queries.md#L15)):*
> ```Markdown
> ordering](query-order.html) or [row limit](limit-offset.html).
> 
> Selection Queries can occur:
> ```

Let's use `Selection queries`, as you've done in body text elsewhere.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-:-L91Z0c52y5OkvbdcOwv:b-8z0szm)*
<!-- Sent from Reviewable.io -->

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/dc60e25d490a555051322dbd584e0d8c4ed83da0/

knz commented:
Review status: 0 of 70 files reviewed at latest revision, 12 unresolved discussions, all commit checks successful.

---

*[v2.0/selection-queries.md, line 11 at r4](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L91YJXQEn5Hw2_cLjwJ:-L92CZWk0Xo8p-gvAwOR:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/16b5ff44f3d0bcef22751b34786a9f511e3c316a/v2.0/selection-queries.md#L11)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

one or more selection clauses
</blockquote></details>

Done.

---

*[v2.0/selection-queries.md, line 15 at r4](https://beta.reviewable.io/reviews/cockroachdb/docs/2762#-L91YWciDnrignWZz-fJ:-L92CbFiBert-1lpz8iF:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/16b5ff44f3d0bcef22751b34786a9f511e3c316a/v2.0/selection-queries.md#L15)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Let's use `Selection queries`, as you've done in body text elsewhere.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/docs/2762)*
<!-- Sent from Reviewable.io -->

knz commented:
Ok fixed. I surmise we can merge this now?

jseldess commented:
Yes. Thank you, @knz!
